includes/admin/upgrades/v3/class-data-migrator.php

```diff
@@ -581,8 +591,11 @@ public static function orders( $data = null ) {
 		unset( $customer_address_data['first_name'] );
 		unset( $customer_address_data['last_name'] );
 
-		// Maybe add address to customer record.
-		edd_maybe_add_customer_address( $customer_id, $customer_address_data );
+		// Maybe add address to customer record, if it is not empty.
+		$customer_address_data = array_filter( $customer_address_data );
```

Comment:
https://github.com/easydigitaldownloads/easy-digital-downloads/pull/7906/files#diff-15bee25f82f2eaea59129e303e1bbf13R643 will take care of this a little later down the pipe, so I don't think this extra check will be required going forward.

Some discussion related to that is here: https://github.com/easydigitaldownloads/easy-digital-downloads/pull/7895#issuecomment-629185205